share/shwap/range_namespace_data.go

```diff
@@ -390,9 +326,132 @@ func RangeCoordsFromIdx(edsIndex, length, edsSize int) (SampleCoords, SampleCoor
 	return from, toCoords, nil
 }
 
-func generateSharesProofs(
+func computeRoot(
+	shares []libshare.Share,
+	ns libshare.Namespace,
+	proof *nmt.Proof, nsCompleteness bool,
+) ([]byte, error) {
+	// this is an expected behavior.
+	if proof == nil {
+		return nil, nil
+	}
+
+	if len(shares) == 0 {
+		return nil, errors.New("empty share list")
+	}
+
+	nth := nmt.NewNmtHasher(
+		share.NewSHA256Hasher(),
+		nmt_ns.ID(ns.Bytes()).Size(),
+		true,
+	)
+
+	if proof.IsEmptyProof() || proof.IsOfAbsence() {
+		return nil, fmt.Errorf("incomplete row proof is invalid")
+	}
+
+	// Compute leaf hashes for the namespace merkle tree
+	hashes, err := nmt.ComputePrefixedLeafHashes(nth, ns.Bytes(), libshare.ToBytes(shares))
+	if err != nil {
+		return nil, fmt.Errorf("failed to compute leaf hashes: %w", err)
+	}
+	// Compute the row root using the namespace merkle tree proof
+	return proof.ComputeRootWithBasicValidation(nth, ns.Bytes(), hashes, nsCompleteness)
+}
+
+func buildTreeRootFromLeaves(shares [][]byte, index uint) ([]byte, error) {
```

Comment:
i prefer to have the parameter named more accurately especially when it is being used to compute somethings. e.g., `extendedShares` here as later tree size is determined as `len(shares)/2)`.
same applies in other places. i would leave it `shares` if no such assumptions (of being extended / original) is made in the function.